### ISSUE TITLE
Model not converging on multiple tasks for Robocasa Dataset.

### ISSUE THREAD

**Issue by ChinChinati** [Physical-Intelligence/openpi#612]
Thanks for the great work. 

We were trying to fine-tune Pi0-FAST and Pi0-FAST-DROID model checkpoints on Robocasa dataset. But we couldn't get the model to work well when trained on multiple tasks. Here's a summary:

| Experiment | Base Model     | # Tasks | Episodes per Task | Total Episodes | Batch Size | Train Steps | Effective Samples Seen (batch × steps) | Success Rate    |
| ---------- | -------------- | ------- | ----------------- | -------------- | ---------- | ----------- | -------------------------------------- | --------------- |
| *Exp 1*  | Pi0-FAST-DROID | 1       | 1500              | 1500           | 32         | 30,000      | 960,000                                | *42%*         |
| *Exp 1′* | Pi0-FAST       | 1       | 1500              | 1500           | 32         | 30,000      | 960,000                                | *35%*         |
| *Exp 2*  | Pi0-FAST       | 3       | 1500              | 4500           | 32         | 90,000      | 2,880,000                              | *20% (avg)*   |
| *Exp 3*  | Pi0-FAST-DROID | 6       | 1500              | 9000           | 32         | 180,000     | 5,760,000                              | *0.08% (avg)* |

For Exp 1 the success rate is acceptable as a baseline policy performance on that task is just 18% and the behaviour isn't bad as well. But for Exp 2 and Exp 3, seems like the robot is barely doing anything meaningful.

For all Exp we have batch_size = 32 and default cosine scheduled LR with peak = 2.5e-5, decays to 2.5e-6 in 30_000 steps (same as Libero Training setup). While we noticed that we are training on much more data than Libero and in the repo for DROID training we had batch_size = 256 and LR was constant throughout the training, with value = 5e-5. Here's how the Schedule looks like: 

<img width="1565" height="882" alt="Image" src="https://github.com/user-attachments/assets/8d25f65d-78a8-45d4-b522-7651bd3fee68" />

I suspect the main reason for bad performance on multiple tasks is that the LR decays too quickly for longer experiments? Since each training run is quite expensive and time consuming, can you give your thought on the matter? 

Can provide more details about the experiments. Thanks in advance. @kpertsch

**Comment by kpertsch** [comment]
I would suggest increasing the batch size to 256 for the larger dataset and setting a constant learning rate of 5e-5! Also you can start from pi05 now!

**Comment by kylekam** [comment]
@ChinChinati were you able to resolve this? I'm noticing similar behavior even with pi05, batch size 256, and constant learning rate of 5e-5.
